### PR TITLE
Offloading: ip mapping (virtualkubelet integration)

### PR DESCRIPTION
**Pull request by cheina97**
# Description

This PR enable the automatic replication of remapped **endpointslices** created starting from **IP** resources.

The main introduced features are:
- Implemented a new transformer inside the **virtualkubelet endpointslices reflector**.
- Added the logic to remap **overlapped endpointslices** inside the **shadowendpointslices** operator.

### CODE REVIEW

File: pkg/liqo-controller-manager/shadowendpointslice-controller/mapping.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		_, extnetMapped, err = net.ParseCIDR(cfg.Status.Remote.CIDR.External.String())
+		if err != nil {
+			return endpoints, err
+		}
+		extNetMaskLen, _ = extnetMapped.Mask.Size()
+	}
+
+	if err != nil {
+		return endpoints, err
+	}
```
Comment (fra98) [inline pkg/liqo-controller-manager/shadowendpointslice-controller/mapping.go:69]: Why this additional check?

File: pkg/virtualKubelet/reflection/exposition/endpointslice.go [hunk 2]
```diff
@@ -273,9 +278,27 @@ func (ner *NamespacedEndpointSliceReflector) ShouldUpdateShadowEndpointSlice(ctx
...
+// MapEndpointIPFromIPResource maps an IP string using an IP resource.
+func (ner *NamespacedEndpointSliceReflector) MapEndpointIPFromIPResource(original string) (string, error) {
+	ips, err := ner.localIPs.List(labels.Everything())
+	if err != nil {
+		return "", fmt.Errorf("failed to list IPs: %w", err)
+	}
+	for i := range ips {
+		if ips[i].Spec.IP.String() == original {
+			if len(ips[i].Status.IPMappings) > 0 {
+				return remapping.GetFirstIPFromMapping(ips[i].Status.IPMappings), nil
```
Comment (fra98) [inline pkg/virtualKubelet/reflection/exposition/endpointslice.go:290]: This function seems unexported by the package

File: pkg/liqo-controller-manager/shadowendpointslice-controller/shadowendpointslice_controller.go
Comment (fra98) [inline pkg/liqo-controller-manager/shadowendpointslice-controller/shadowendpointslice_controller.go:1]: Should we watch for Configuration updates if CIDRs changes?

Comment (cheina97) [reply pkg/liqo-controller-manager/shadowendpointslice-controller/shadowendpointslice_controller.go:1]: We assume that CIDRs never change

**Pull request merged**